internal/cli/pkg/cluster/util.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright (c) 2024 OceanBase
+ob-operator is licensed under Mulan PSL v2.
+You can use this software according to the terms and conditions of the Mulan PSL v2.
+You may obtain a copy of Mulan PSL v2 at:
+
+	http://license.coscl.org.cn/MulanPSL2
+
+THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+See the Mulan PSL v2 for more details.
+*/
+package cluster
+
+import (
+	"crypto/rand"
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+
+	apitypes "github.com/oceanbase/ob-operator/api/types"
+
+	"github.com/oceanbase/ob-operator/internal/dashboard/model/common"
+
+	param "github.com/oceanbase/ob-operator/internal/dashboard/model/param"
+)
+
+func generateUserSecrets(clusterName string, clusterId int64) *apitypes.OBUserSecrets {
+	return &apitypes.OBUserSecrets{
+		Root:     fmt.Sprintf("%s-%d-root-%s", clusterName, clusterId, generateUUID()),
+		ProxyRO:  fmt.Sprintf("%s-%d-proxyro-%s", clusterName, clusterId, generateUUID()),
+		Monitor:  fmt.Sprintf("%s-%d-monitor-%s", clusterName, clusterId, generateUUID()),
+		Operator: fmt.Sprintf("%s-%d-operator-%s", clusterName, clusterId, generateUUID()),
+	}
+}
+
+func CheckResourceName(name string) bool {
+	// 定义正则表达式
+	regex := `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`
+
+	// 编译正则表达式
+	re, err := regexp.Compile(regex)
+	if err != nil {
+		log.Println("Error compiling regex:", err)
+		return false
+	}
+
+	// 检查整个字符串是否符合正则表达式的模式
+	return re.MatchString(name)
+}
+func CheckPassword(password string) bool {
+	var (
+		countUppercase   int
+		countLowercase   int
+		countNumber      int
+		countSpecialChar int
+	)
+
+	// 遍历密码中的每个字符
+	for _, char := range password {
+		switch {
+		case regexp.MustCompile(`[A-Z]`).MatchString(string(char)):
+			countUppercase++
+		case regexp.MustCompile(`[a-z]`).MatchString(string(char)):
+			countLowercase++
+		case regexp.MustCompile(`[0-9]`).MatchString(string(char)):
+			countNumber++
+		default:
```

Comment:
White space characters will also be counted here. Maybe a determined list of special character instead of `default` is better here.

Reply:
Yes,  I thick maybe use `[^\w\s]` instead?